src/components/README.md

```diff
@@ -0,0 +1,49 @@
+# React components - `/src/components/`
+
+This folder contains all of the React components that power perf.html. The folder structure is broken down by the contents of different tabs, and other logical groupings. For instance `calltree` and `timeline` contain the components that power the call tree view, and the timeline view respectively. These components can be both connected components (e.g. components that are actively connected to the Redux store), utility components, and components that are directly passed in props. Making components generic is fine, but they are still stored close to where they are used. The only exception is when a components starts to be shared across multiple areas. In this case it is located in the `shared` folder. The preference for modules should be to keep them close to where they are used, even if it would be easy to abstract them out.
+
+## Pure components
+
+This project assumes that the data it uses is immutable. Because of this, each component should be a PureComponent to reduce the amount of component updates that happen for each individual mounted component. Pure components are components that use shallow equality on the props to determine whether the component should be updated. Effort should be made to ensure that components will do the right thing when performing this equality check. Derived data should be memoized whenever possible so that `===` checks work correctly.
+
+## Connecting Redux
+
+Some components use the `connect()` function to hook up the contents of the [Redux](http://redux.js.org/) store to be used within the component, and to wrap the action creators with the store's dispatch method.
+
+### Example connect:
+
+```js
+export default connect(
+  // The first parameter's function receives the Redux state. Use the selectors located
+  // in the reducers directory to select the relevant information from the state.
+  // Selectors are often memoized, and so as these functions are run multiple times,
+  // their results will still pass a strict equality test for component updates. In
+  // addition notice how the state object is Flow typed.
```

Comment:
actually, react-redux' `connect` memoizes stuff by itself too. In the end I think we end up doing many unneeded equality checks everywhere -- but they're cheap so maybe it's not a big deal.